### PR TITLE
stdlib: add functions for tuple types

### PR DESCRIPTION
**Pull request by michaelpj**
Fairly self-explanatory.

### CODE REVIEW

File: language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Tuple.hs [hunk 1]
```diff
@@ -0,0 +1,101 @@
+-- | @tuple@s of various sizees and related functions.
+
+{-# LANGUAGE OverloadedStrings #-}
+
+module Language.PlutusCore.StdLib.Data.Tuple
```
Comment (effectfully) [inline language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Tuple.hs]: I'd like to keep the internal stdlib modules purely PLCish. Can we move this to `Language.PlutusCore.StdLib.Meta.Data.Tuple` or the like? Maybe even `Language.PlutusCore.Tuple`? I mean, those functions are library functions, but for Haskell users, not PLC users.

Comment (michaelpj) [reply language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Tuple.hs]: Well, we do have `fixN` in `Language.PlutusCore.Stdlib.Data.Function`. Probably that should go in `Language.PlutusCore.Stdlib.Data.Function` as well?

Comment (michaelpj) [reply language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Tuple.hs]: I'm also not sure how to include those nicely in the testing - at the moment I've arbitrarily put in instantiations at 2 (like we do for `fixN`)

File: language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Tuple.hs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import           Language.PlutusCore.MkPlc
+import           Language.PlutusCore.Name
+import           Language.PlutusCore.Quote
+import           Language.PlutusCore.Renamer
+import           Language.PlutusCore.Type
+
+import           Data.Traversable
+
+-- | Given an arity @n@, create the type of n-tuples.
+-- > \(T_1 :: *) .. (T_n :: *) . all (R :: *) . (T_1 -> .. -> T_n -> R) -> R
```
[the remark this answers is not in the document]
Comment (michaelpj) [reply language-plutus-core/stdlib/Language/PlutusCore/StdLib/Data/Tuple.hs]: I should probably actually look at the haddock output...

**Pull request merged**